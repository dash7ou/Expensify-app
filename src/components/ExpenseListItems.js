import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseItem = ({ description ,id, amount , createdAt})=>(

        <Link className="list-items" to={`/edit/${id}`}>
            <div>
                {description && <h3 className="list-items__title">{description}</h3>}
                <span className="list-item__sub-title"> {moment(createdAt).format('MMMM Do, YYYY')} </span>
            </div>
            <h3 className="list-item__data"> {numeral(amount /100).format('$0,0.00')} </h3>
        </Link>

)

export default ExpenseItem;