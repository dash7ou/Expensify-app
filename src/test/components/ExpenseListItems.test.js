import React from "react";
import { shallow } from "enzyme";
import ExpenseListItems from "../../components/ExpenseListItems";

// data

const expenses = [
    {
        id: "1",
        description: "dash",
        amount: 1000,
        createdAt: 0,
        note: ""
    }
]

test("should render ExpenseListItems correctly", ()=>{
    const wrapper = shallow(<ExpenseListItems expense={expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
})