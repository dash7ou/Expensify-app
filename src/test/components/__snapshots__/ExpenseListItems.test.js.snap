// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItems correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/undefined"
  />
  <p>
     - 
  </p>
</div>
`;
